core/ir/x86/instr_create_api.h

```diff
@@ -1578,6 +1578,11 @@
 #define INSTR_CREATE_bndmk(dc, d, s) instr_create_1dst_1src((dc), OP_bndmk, (d), (s))
 #define INSTR_CREATE_bndldx(dc, d, s) instr_create_1dst_1src((dc), OP_bndldx, (d), (s))
 #define INSTR_CREATE_bndstx(dc, d, s) instr_create_1dst_1src((dc), OP_bndstx, (d), (s))
+/* MOVDIRI */
+#define INSTR_CREATE_movdiri(dc, d, s) instr_create_1dst_1src((dc), OP_movdiri, (d), (s))
+/* MOVDIR64B */
+#define INSTR_CREATE_movdir64b(dc, d, s) \
```

Comment:
Could be nice to ensure drutil_insert_get_mem_addr() works inside the drmemtrace framework: it should convert the base-disp into a lea (which should always work?); the multiple memrefs should be handled.  Optional; ideal thing would be inside a drmemtrace test but those generally execute the instructions and that would fail on today's hardware.  Probably not worth it then.

Reply:
Happy to do this in a follow up issue (please file) but I've literally never used drmemtrace before and have no idea how to write a test with it. I have hardware these instructions exist on so I can at least test it manually even if we can't test it in CI.